ui/src/main/js/page/audit/Details.js

```diff
@@ -82,14 +84,29 @@ class Details extends Component {
     }
 
     resendButton(cell, row) {
-        return (
-            <RefreshTableCellFormatter
-                id="audit-detail-refresh-cell"
-                handleButtonClicked={this.onResendClick}
-                currentRowSelected={row}
-                buttonText="Re-send"
-            />
-        );
+        if (this.isResendAllowed()) {
+            return (
+                <RefreshTableCellFormatter
+                    id="audit-detail-refresh-cell"
+                    handleButtonClicked={this.onResendClick}
+                    currentRowSelected={row}
+                    buttonText="Re-send"
+                />
+            );
+        }
+        return (<div className="jobIconButtonDisabled"><FontAwesomeIcon icon="sync" className="alert-icon" size="lg" /></div>);
+    }
+
+    isResendAllowed() {
+        const { descriptors } = this.props;
+        if (descriptors) {
+            const descriptorList = DescriptorUtilities.findDescriptorByNameAndContext(descriptors, DescriptorUtilities.DESCRIPTOR_NAME.COMPONENT_AUDIT, DescriptorUtilities.CONTEXT_TYPE.GLOBAL);
+            if (descriptorList) {
+                return descriptorList.some((descriptor) => DescriptorUtilities.isOperationAssigned(descriptor, DescriptorUtilities.OPERATIONS.EXECUTE));
```

Comment:
I believe you can return
```suggestion
            return descriptorList && descriptorList.some((descriptor) => DescriptorUtilities.isOperationAssigned(descriptor, DescriptorUtilities.OPERATIONS.EXECUTE));
```
it'll verify the first item isn't null and then perform the check